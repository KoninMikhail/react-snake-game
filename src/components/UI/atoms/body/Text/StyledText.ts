import styled from 'styled-components';
import {
  font_color,
  font_family_body,
  font_size_body,
  font_size_body_desktop,
} from '../../../../../styles/variables';
import { up } from 'styled-breakpoints';
import { ITextProps } from './Text';

export const StyledText = styled.p<ITextProps>`
  font-size: ${font_size_body};
  font-family: ${font_family_body};
  font-weight: normal;
  color: ${font_color};

  ${up('xl')} {
    font-size: ${font_size_body_desktop};
  }
`;
