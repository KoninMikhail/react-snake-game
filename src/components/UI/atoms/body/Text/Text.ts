import styled from 'styled-components';
import {
  font_color,
  font_family_body,
  font_size_body,
  font_size_body_desktop,
} from '../../../../../styles/variables';
import { up } from 'styled-breakpoints';

/**
 * Text
 *
 * @description Use for label content
 *
 * @returns {JSX.Element}
 */
export const Text = styled.p`
  font-size: ${font_size_body};
  font-family: ${font_family_body};
  font-weight: normal;
  color: ${font_color};

  ${up('xl')} {
    font-size: ${font_size_body_desktop};
  }
`;
