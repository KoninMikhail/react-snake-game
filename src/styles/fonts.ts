// fonts.ts
import { createGlobalStyle } from 'styled-components';
import PixeloidBoldWoff2 from 'static/fonts/pixeloidsans-bold-webfont.woff2';
import PixeloidBoldWoff from 'static/fonts/pixeloidsans-bold-webfont.woff';

export const FontStyles = createGlobalStyle`
  @font-face {
    font-family: 'pixeloid_bold';
    font-weight: normal;
    font-style: normal;
    src: url(${PixeloidBoldWoff2}) format('woff2'),
    url(${PixeloidBoldWoff}) format('woff');
  }
`;
