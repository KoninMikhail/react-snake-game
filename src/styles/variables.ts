/**
 * Wrapper
 */
export const wrapper = '1.5rem';
export const wrapper_desktop = '5vh 4vw';

/**
 * Typography
 */
export const font_size_body = '14px';
export const font_size_body_desktop = '1vw';
export const font_family_body = `'pixeloid_bold','Segoe UI','Roboto','Oxygen','Ubuntu','Fira Sans','Droid Sans','Helvetica Neue',sans-serif`;
export const font_color = 'white';

/**
 * StyledGrid
 */

export const gap = '1rem';
