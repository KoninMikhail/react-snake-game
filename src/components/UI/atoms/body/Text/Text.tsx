import { StyledText } from './StyledText';
import { ElementType } from 'react';

export interface ITextProps {
  as?: ElementType;
  children?: JSX.Element | JSX.Element[] | string;
}

/**
 * Text
 *
 * Use for add text content
 *
 * @returns {JSX.Element}
 */

export const Text = ({ as, children }: ITextProps) => <StyledText as={as}>{children}</StyledText>;
