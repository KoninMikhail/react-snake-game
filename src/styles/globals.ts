// global.ts
import { createGlobalStyle } from 'styled-components';
import { font_color, font_family_body, font_size_body, font_size_body_desktop } from './variables';
import { up } from 'styled-breakpoints';

export const GlobalStyles = createGlobalStyle`
  * {
    box-sizing: border-box;
  }

  html {
    margin: 0;
    height: 100%;
  }

  body {
    margin: 0;
    width: 100%;
    font-size: ${font_size_body};
    height: 100%;
    color: ${font_color};
    background-color: black;
    font-family: ${font_family_body};
    -webkit-font-smoothing: antialiased;
    -moz-osx-font-smoothing: grayscale;
    user-select: none;
    overscroll-behavior: contain;

    ${up('xl')} {
      font-size: ${font_size_body_desktop};
    }
  }

  #root {
    width: inherit;
    height: inherit;
    display: block;
    position: relative;
  }
`;
