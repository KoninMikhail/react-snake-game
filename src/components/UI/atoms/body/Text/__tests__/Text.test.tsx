import { render, screen } from '@testing-library/react';
import { Text } from '../Text';
import { font_color, font_family_body, font_size_body } from '../../../../../../styles/variables';

describe('Text', () => {
  it('render a text element', () => {
    render(<Text>Awesome text</Text>);
    expect(screen.getByText(/Awesome text/i)).toBeInTheDocument();
  });

  it('have correctly styles', () => {
    render(<Text>Awesome text</Text>);

    expect(screen.getByText(/Awesome text/i)).toHaveStyleRule('font-size', font_size_body);
    expect(screen.getByText(/Awesome text/i)).toHaveStyleRule('font-family', font_family_body);
    expect(screen.getByText(/Awesome text/i)).toHaveStyleRule('color', font_color);
  });

  it('can be render with another tag-name', () => {
    render(<Text as="span">Awesome text</Text>);

    expect(screen.getByText(/Awesome text/i).tagName.toLowerCase()).toEqual('span');
  });

  it('equal snapshot', () => {
    const { container } = render(<Text>Awesome text</Text>);
    expect(container).toMatchSnapshot();
  });
});
