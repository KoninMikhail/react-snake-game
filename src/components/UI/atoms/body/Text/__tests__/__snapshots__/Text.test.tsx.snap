// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Text equal snapshot 1`] = `
.c0 {
  font-size: 14px;
  font-family: 'pixeloid_bold','Segoe UI','Roboto','Oxygen','Ubuntu','Fira Sans','Droid Sans','Helvetica Neue',sans-serif;
  font-weight: normal;
  color: white;
}

@media (min-width:1200px) {
  .c0 {
    font-size: 1vw;
  }
}

<div>
  <p
    class="c0"
  >
    Awesome text
  </p>
</div>
`;
